core/types/types.cc

```diff
@@ -843,17 +848,12 @@ core::SymbolRef Types::getRepresentedClass(core::Context ctx, const core::Type *
         singleton = s->symbol;
     } else {
         auto *at = core::cast_type<core::AppliedType>(ty);
-        if (at == nullptr) {
-            return core::Symbols::noSymbol();
-        }
-        // If this class is a "real" generic, leave it alone. We're not quite
-        // sure yet what it means to move between the "attached" and "singleton"
-        // levels for generic singletons. However, if the singleton has kind `*`
-        // due to all type members being fixed:, we're OK to treat the
-        // AppliedType and the Symbol as interchangeable here.
-        if (at->klass.data(ctx)->typeArity(ctx) != 0) {
+
+        // TODO: should the singletonClass check be an ENFORCE?
+        if (at == nullptr || !at->klass.data(ctx)->isSingletonClass(ctx)) {
```

Comment:
what is the `!at->klass.data(ctx)->isSingletonClass(ctx)` for?
it seems super surprising to have it in `AppliedType` case but not `ClassType` case.

Reply:
I get the feeling that it's just not necessary. I had originally added it in because it wasn't obvious to me that we wouldn't call this function with a non-singleton class.

Reply:
So I've decided that the check just wasn't necessary, and removed it. The tests are all fine, and I think that my initial thinking about what sorts of types get passed to `getRepresentedClass` was just wrong.